diag_manager/fms_diag_yaml.F90

```diff
@@ -68,17 +69,15 @@ module fms_diag_yaml_mod
 
 !> @brief type to hold the sub region information about a file
 type subRegion_type
-  character (len=:), allocatable :: grid_type !< Flag indicating the type of region,
-                                              !! acceptable values are "latlon" and "index"
-  real, allocatable :: lat_lon_sub_region (:) !< Array that stores the grid point bounds for the sub region
-                                              !! to use if grid_type is set to "latlon"
-                                              !! [dim1_begin, dim1_end, dim2_begin, dim2_end,
-                                              !!  dim3_begin, dim3_end, dim4_begin, dim4_end]
-  integer, allocatable :: index_sub_region (:) !< Array that stores the index bounds for the sub region to
-                                               !! to use if grid_type is set to "index"
-                                               !! [dim1_begin, dim1_end, dim2_begin, dim2_end,
-                                               !!  dim3_begin, dim3_end, dim4_begin, dim4_end]
-  integer :: tile !< Tile number of the sub region, required if using the "index" grid type
+  character (len=:), allocatable :: grid_type   !< Flag indicating the type of region,
```

Comment:
I think you should save this as an integer, and set an integer parameter to represent each acceptable value.  Then when you do the comparison, you will be comparing integers instead of strings.
```fortran
integer, parameter :: latlon_gridtype = 1
integer, paramerer :: index_gridtype = 2
integer, paramerer :: null_gridtype = DIAG_NULL
```
```fortran
! Or use if below
select case (grid)
 case(latlon_gridtype)
!  latlon stuff
 case (index_gridtype)
! index stuff
 case (null_gridtype)
! error?
 case default
! This was never set
end select
```